lib/private/Activity/Manager.php

```diff
@@ -262,15 +263,17 @@ public function registerSetting(string $setting): void {
 	}
 
 	/**
-	 * @return ISetting[]
+	 * @return ActivitySettings[]
 	 * @throws \InvalidArgumentException
 	 */
 	public function getSettings(): array {
 		foreach ($this->settingsClasses as $class => $false) {
 			/** @var ISetting $setting */
 			$setting = \OC::$server->query($class);
 
-			if (!$setting instanceof ISetting) {
+			if ($setting instanceof ISetting) {
```

Comment:
With the recent change this needs to be adjusted as well:

```suggestion
			if ($setting instanceof ISetting && !$setting instanceof ActivitySettings) {
```

Or being moved into an outer if statement to make it better readable.

Reply:
done